config/jobs/kubernetes/kops/build-pipeline.py

```diff
@@ -0,0 +1,125 @@
+      args:
+      # Generic e2e test args
+      - --up
+      - --test
+      - --down
+      - --dump=$(ARTIFACTS)
+      - --timeout=45m
+      - --gcp-service-account=$(E2E_GOOGLE_APPLICATION_CREDENTIALS)
+      # kops-specific test args
+      - --deployment=kops
+      - --provider=aws
+      - --cluster={{name}}.test-cncf-aws.k8s.io
+      - --kops-ssh-user={{ssh_user}}
+      - --kops-nodes=4
+      - --extract={{extract}}
+      - --kops-state=s3://k8s-kops-prow/
+      - --kops-ssh-key=$(AWS_SSH_PRIVATE_KEY_FILE)
+      - --kops-ssh-public-key=$(AWS_SSH_PUBLIC_KEY_FILE)
+      - --kops-publish=gs://k8s-staging-kops/kops/releases/markers/{{branch}}/latest-ci-updown-green.txt
+      - --kops-version=https://storage.googleapis.com/k8s-staging-kops/kops/releases/markers/{{branch}}/latest-ci.txt
+      #- --kops-kubernetes-version should be inferred by kubetest from --extract
+      #- --kops-zone should be randomized by kubetest
+      # Specific test args
+      - --test_args=--ginkgo.focus=\\[k8s.io\\]\\sNetworking.*\\[Conformance\\]
+      - --ginkgo-parallel
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-kops, google-aws, kops-misc, kops-k8s-{{k8s_version}}
+    testgrid-tab-name: {{tab}}
+"""
+
+def build_tests(branch, k8s_version, ssh_user):
+    def expand(s):
+        subs = {}
+        if k8s_version:
+            subs['k8s_version'] = k8s_version
+        if branch:
+            subs['branch'] = branch
+        return s.format(**subs)
+
+    if branch == 'master':
+        extract = "release/latest"
+        e2e_image = "gcr.io/k8s-testimages/kubekins-e2e:v20200602-05eeaff-master"
+    else:
+        extract = expand("release/stable-{k8s_version}")
+        # Hack to stop the autobumper getting confused
+        e2e_image = "gcr.io/k8s-testimages/kubekins-e2e:v20200602-05eeaff-1.18"
+        e2e_image = e2e_image[:-4] + k8s_version
+
+    tab = expand('kops-pipeline-updown-{branch}')
+
+    # Names must be valid pod and DNS names
+    name = expand('e2e-kops-pipeline-updown-kops{branch}')
+    name = name.replace('.', '')
+
+    y = template
+    y = y.replace('{{branch}}', branch)
+    y = y.replace('{{extract}}', extract)
+    y = y.replace('{{e2e_image}}', e2e_image)
+    y = y.replace('{{k8s_version}}', k8s_version)
+    y = y.replace('{{name}}', name)
+    y = y.replace('{{ssh_user}}', ssh_user)
+    y = y.replace('{{tab}}', tab)
+
+    spec = {
+        'branch': branch,
+        'k8s_version': k8s_version,
+    }
+    jsonspec = json.dumps(spec, sort_keys=True)
+
+    print("")
+    print("# " + jsonspec)
+    print(y.strip())
+
+
+branches = [
+    "master",
+    "1.16",
+    "1.17",
+    "1.18",
+]
+
+def generate():
+    print("# Test scenarios generated by build-pipeline.py (do not manually edit)")
+    print("periodics:")
+    for branch in branches:
+        k8s_version = "latest" if branch == "master" else branch
+        ssh_user = "ubuntu" if branch in ("master", "1.18") else "admin"
```

Comment:
should we invert this logic so that we don't need to modify it for every new kops version?

```suggestion
        ssh_user = "admin" if branch in ("1.16", "1.17") else "ubuntu"
```

then we can just remove it when we stop testing 1.16 and 1.17